master/internal/rm/kubernetesrm/nodes.go

```diff
@@ -1,101 +1,71 @@
 package kubernetesrm
 
 import (
-	"github.com/determined-ai/determined/master/pkg/actor"
+	"context"
 
+	"github.com/sirupsen/logrus"
 	k8sV1 "k8s.io/api/core/v1"
-	k8Informers "k8s.io/client-go/informers"
-	k8sClient "k8s.io/client-go/kubernetes"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
 )
 
-// Messages sent by the nodeInformer to itself.
-type (
-	startNodeInformer struct{}
-)
-
-// Messages sent by the nodeInformer to the podsHandler.
-type (
-	nodeStatusUpdate struct {
-		updatedNode *k8sV1.Node
-		deletedNode *k8sV1.Node
-	}
-)
+type nodeCallbackFunc func(*k8sV1.Node, watch.EventType)
 
 type nodeInformer struct {
-	informer    k8Informers.SharedInformerFactory
-	podsHandler *actor.Ref
-	stop        chan struct{}
+	cb            nodeCallbackFunc
+	nodeInterface typedV1.NodeInterface
+	syslog        *logrus.Entry
+	resultChan    <-chan watch.Event
 }
 
-func newNodeInformer(clientSet k8sClient.Interface, podsHandler *actor.Ref) *nodeInformer {
-	return &nodeInformer{
-		informer: k8Informers.NewSharedInformerFactoryWithOptions(
-			clientSet, 0, []k8Informers.SharedInformerOption{}...),
-		podsHandler: podsHandler,
-		stop:        make(chan struct{}),
+func newNodeInformer(
+	ctx context.Context,
+	nodeInterface typedV1.NodeInterface,
+	cb nodeCallbackFunc,
+) (*nodeInformer, error) {
+	nodes, err := nodeInterface.List(ctx, metaV1.ListOptions{})
+	if err != nil {
+		return nil, err
 	}
-}
 
-func (n *nodeInformer) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		ctx.Tell(ctx.Self(), startNodeInformer{})
-
-	case startNodeInformer:
-		if err := n.startNodeInformer(ctx); err != nil {
-			return err
-		}
-
-	case actor.PostStop:
-		ctx.Log().Infof("shutting down node informer")
-		close(n.stop)
-
-	default:
-		ctx.Log().Errorf("unexpected message %T", msg)
-		return actor.ErrUnexpectedMessage(ctx)
+	rw, err := watchtools.NewRetryWatcher(nodes.ResourceVersion, &cache.ListWatch{
+		WatchFunc: func(options metaV1.ListOptions) (watch.Interface, error) {
+			return nodeInterface.Watch(ctx, options)
+		},
+	})
+	if err != nil {
+		return nil, err
 	}
-
-	return nil
+	for _, node := range nodes.Items {
+		cb(&node, watch.Added)
```

Comment:
linter catches it, but this is a loop capture bug. go is going to break the language to fix how deceptive this is, hopefully (https://github.com/golang/go/issues/57969).